docs/src/explanations/chisel-enum.md

```diff
@@ -0,0 +1,134 @@
+# ChiselEnum
+
+The ChiselEnum type can be used to reduce the chance of error when encoding mux selectors, opcodes, and functional unit operations. In contrast with`Chisel.util.Enum`, `ChiselEnum` are subclasses of `Data`, which means that they can be used to define fields in `Bundle`s, including in `IO`s.
+
+
+## Functionality and Examples
+
+```scala mdoc
+// Imports used in the following examples
+import chisel3._
+import chisel3.util._
+import chisel3.stage.ChiselStage
+import chisel3.experimental.ChiselEnum
+```
+
+Below we see ChiselEnum being used as mux select for a RISC-V core. While wrapping the object in a package is not required, it is highly recommended as it allows for the type to be used in multiple files more easily. 
+
+```scala mdoc
+// package CPUTypes {
+    object AluMux1Sel extends ChiselEnum {
+        val selectRS1, selectPC = Value
+        /**
+            "selectRS1" -> 0.U,
+            "selectPC"  -> 1.U
+        */
+    }
+// }
+```
```

Comment:
To narrowly document `ChiselEnum`, it's sufficient to ignore suggestions of how a user should organize/package their code.

That said, if you want to show this as living under `CPUTypes`, I think that's reasonable. Because Scala seemingly treats objects and packages as basically the same, you can show what you want with:
```scala
object CPUTypes {
  object AluMux1Sel extends Chisel Enum { /** */ }
}
```

Reply:
Additionally, it would be good to make clear what the `"selectRS1" -> 0.U` portion is supposed to be doing and what that does. It would be fine to show this with two different variants of `AluMux1Sel`, e.g., `AluMux1Sel` and `AluMex1SelExplicitValues` or something.

Reply:
Right so this the mapping comment comes from a misunderstanding I had with the value assignment/verification code in the example StrongEnum code. This is the actual code that I have in my file

```scala
object AluMux1Sel extends ChiselEnum {
        val rs1out, pcout = Value
        val correct_annotation_map = Map[String, UInt](elems =
            "rs1out" -> 0.U,
            "pcout"  -> 1.U
        )
    }
```

When I first saw this mapping function, I thought the mapping is what actually assigned the value BUT IT DOESN'T!!! For the documentation I removed the actual function because it is only for verification and not for assigning value.

Reply:
i think the current form of the doc is helpful, I found it clear what it was saying

Reply:
Ah, I get it now. Maybe just extend the comment to say: "This is what the mapping will be" or something. Not blocking. I'm fine with it the way it is. :+1: